src/store.js

```diff
@@ -0,0 +1,260 @@
+}
+
+///////////////////////////////////////////////////////////////////////////
+////////////////////////     REDUX - REDUCERS    //////////////////////////
+///////////////////////////////////////////////////////////////////////////
+
+////////////////////////   REDUX - AUTH REDUCER   ////////////////////
+
+const authReducer = (state = {}, action) => {
+  if (action.type === SET_AUTH) {
+    state = action.auth;
+  }
+  return state;
+};
+
+////////////////////////   REDUX - PRODUCTS REDUCER   ////////////////////
+
+const productReducer = (state = [], action) => {
+  if (action.type === SET_PRODUCTS) {
+    state = action.products;
+  }
+  return state;
+};
+
+////////////////////////   REDUX - USERS REDUCER   ////////////////////
+
+const userReducer = (state = [], action) => {
+  if (action.type === SET_USERS) {
+    state = action.users;
+  }
+  if (action.type === UPDATE_USER) {
+    return state.map(user => action.id === user.id ? {
+      ...user, username: action.username, email: action.email, password: action.password, firstName: action.firstName, lastName: action.lastName, streetAddress: action.streetAddress, city: action.city, state: action.state, zipcode: action.zipcode, billStreetAddress: action.billStreetAddress, billCity: action.billCity, billState: action.billState, billZipcode: action.billZipcode
+    } : user);
+  }
+  return state;
+};
+
+////////////////////////   REDUX - ORDERS REDUCER   ////////////////////
+
+const orderReducer = (state = [], action) => {
+  if (action.type === SET_ORDERS) {
+    state = action.orders;
+  }
+  return state;
+};
+
+const orderProdutsReducer = (state = [], action) => {
+  if (action.type === SET_ORDERPRODUCTS) {
+    state = action.orderProducts;
+  }
+  if (action.type === DELETE_ORDERPRODUCT) {
+    state = state.filter(item => item.id !== action.id)
+  }
+  if (action.type === UPDATE_ORDERPRODUCT) {
+    state = state.map(item => item.id === action.item.id ? action.item : item)
+  }
+  if (action.type === ADD_ORDERPRODUCT) {
+    state = [...state, action.orderProduct]
+  }
+  return state;
+};
+
+////////////////////////   REDUX - COMBINE REDUCERS    ////////////////////
+const reducer = combineReducers({
+  auth: authReducer,
+  products: productReducer,
+  users: userReducer,
+  orders: orderReducer,
+  orderProducts: orderProdutsReducer
+});
+
+//////////////////////////////////////////////////////////////////////////
+////////////////////////  REDUX - CREATE STORE  ///////////////////////////
+///////////////////////////////////////////////////////////////////////////
+
+//REDUX STORE WITH MIDDLEWARE
+const store = createStore(reducer, applyMiddleware(thunk));
+
+export default store;
+export { setProductsAction, setUsersAction };
+export {
+  setProductsThunk,
+  setUsersThunk,
+  updateUserThunk,
+  setOrdersThunk,
+  setOrderProductsThunk,
+  deleteOrderProductsThunk,
+  addOrderProductThunk,
+  updateOrderProductThunk,
+  attemptLogin,
+  attemptSessionLogin,
+  logout
+};
```

Comment:
Crazy thought,

Maybe we can break this file into many files with the names you so kindly gave the comments.